addons/website_slides/static/src/js/slides_course_page.js

```diff
@@ -0,0 +1,148 @@
+            slideId: slide.id,
+            slideCompleted: completed,
+            canSelfMarkUncompleted: slide.canSelfMarkUncompleted,
+            canSelfMarkCompleted: slide.canSelfMarkCompleted,
+            isMember: slide.isMember,
+        });
+
+        if ($button.length) {
+            $button.replaceWith(newButton);
+            const $elem = this.$(`.o_wslides_sidebar_done_button[data-id="${slide.id}"]`);
+            $elem.data('completed', completed);
+        }
+    },
+
+    /**
+     * Updates the progressbar whenever a lesson is completed
+     *
+     * @public
+     * @param {*} channelCompletion
+     */
+    updateProgressbar: function (channelCompletion) {
+        const completion = Math.min(100, channelCompletion);
+
+        const $completed = $('.o_wslides_channel_completion_completed');
+        const $progressbar = $('.o_wslides_channel_completion_progressbar');
+
+        if (completion < 100) {
+            // Hide the "Completed" text and show the progress bar
+            $completed.addClass('d-none');
+            $progressbar.removeClass('d-none').addClass('d-flex');
+        } else {
+            // Hide the progress bar and show the "Completed" text
+            $completed.removeClass('d-none');
+            $progressbar.addClass('d-none').removeClass('d-flex');
+        }
+
+        $progressbar.find('.progress-bar').css('width', `${completion}%`);
+        $progressbar.find('.o_wslides_progress_percentage').text(completion);
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Once the completion conditions are filled,
+     * rpc call to set the relation between the slide and the user as "completed"
+     *
+     * @private
+     * @param {Integer} slideId: the id of slide to set as completed
+     * @param {Boolean} completed: true to mark the slide as completed
+     *     false to mark the slide as not completed
+     */
+    _toggleSlideCompleted: async function (slide, completed = true) {
+        if (!!slide.completed === !!completed || !slide.isMember) {
+            // no useless RPC call
+            return;
+        }
+
+        const data = await this._rpc({
+            route: `/slides/slide/${completed ? 'set_completed' : 'set_uncompleted'}`,
+            params: {slide_id: slide.id},
+        });
+
+        this.toggleCompletionButton(slide, completed);
+        this.updateProgressbar(data.channel_completion);
+
+        slide.completed = true;
+    },
+
+    _getSlide: function (slideId) {
+        return $(`.o_wslides_sidebar_done_button[data-id="${slideId}"]`).data();
+    },
+
+    //--------------------------------------------------------------------------
+    // Handler
+    //--------------------------------------------------------------------------
+    _onClickComplete: function (ev) {
```

Comment:
If you have some time, would be nice to add the various tags (`@private`, `@override`, ...) on your new methods docstrings.